tools/ci_build/build.py

```diff
@@ -518,17 +529,46 @@ def build_python_wheel(source_dir, build_dir, configs, use_cuda):
         if is_ubuntu_1604():
             run_subprocess([os.path.join(source_dir, 'rename_manylinux.sh')], cwd=cwd+'/dist')
 
+def build_protoc_for_windows_host(cmake_path, source_dir, build_dir):
+    if not is_windows():
+        raise BuildError('Currently only support building protoc for Windows host while cross-compiling for ARM/ARM64 arch')
+
+    log.info("Building protoc for host to be used in cross-compiled build process")
+    protoc_build_dir = os.path.join(build_dir, 'host_protoc')
+    os.makedirs(protoc_build_dir, exist_ok=True)
+    # Generate step
+    cmd_args = [cmake_path,
```

Comment:
'-A','x64','-T', toolset

They are not necessary, but it's better to have.

Reply:
I thought about this but (I haven't tested it yet) what if we were cross-compiling on an x86 machine ? In that case using -A x64 -T host=x64 is fatal. The way it currently is it should default to the architecture of the machine it is cross-compiling on. What do you think about this ?

Reply:
"-A x86 -T host=x86"

Reply:
By default, just use x64 toolset for both x64 and x86 and I will let architecture default based on machine for now.